lib/generate/collectModules.js

```diff
@@ -112,5 +114,111 @@ module.exports = async (page) => {
         return modules;
     }, excludedModules);
 
-    return modules;
+    // Calculate dependencies of each module
+    const dependencyMap = new Map();
+    for (const moduleName in modules) {
+        if (moduleName.includes('!')) {
+            // This module has no dependencies
+            dependencyMap.set(moduleName, {});
+            continue;
+        }
+
+        // Load this module anew to track its dependencies
+        const dependencies = await page.evaluate((moduleName) => {
+            window.define = function (name, dependencies) {
+                if (typeof name !== 'string') {
+                    dependencies = name;
+                }
+
+                if (Array.isArray(dependencies)) {
+                    window.magepack_dependencies = dependencies;
+                }
+            };
+
+            return new Promise((resolve) => {
+                // Forget this module, so it gets loaded anew
+                require.undef(moduleName);
+
+                // Global state isn't nice. Alternate suggestions are welcome.
+                window.magepack_dependencies = [];
+
+                // A new context means no mixins get in the way
+                const localContext = require.s.newContext();
```

Comment:
Breaks if minification is enabled, can fix it easily enough: `localContext.nameToUrl = require.s.contexts._.nameToUrl;`

Reply:
Thanks for this. I was inadvertently testing with a (production) website where minification is disabled. I've added this change/fix now.